mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -659,4 +672,30 @@ mod tests {
         let kes_period = client_res.unwrap().unwrap();
         assert_eq!(407, kes_period);
     }
+
+    #[tokio::test]
+    async fn calculate_kes_period() {
+        let socket_path = create_temp_dir("get_current_kes_period").join("node.socket");
+        let observer = PallasChainObserver::new(socket_path.as_path(), CardanoNetwork::TestNet(10));
+        let current_kes_period = observer
+            .calculate_kes_period(Point::Specific(53536042, vec![1, 2, 3]), 129600)
+            .await
+            .unwrap();
+
+        assert_eq!(413, current_kes_period);
+
+        let current_kes_period = observer
+            .calculate_kes_period(Point::Specific(53600000, vec![1, 2, 3]), 129600)
```

Comment:
Maybe this is a better value to test:
```suggestion
            .calculate_kes_period(Point::Specific(53524800, vec![1, 2, 3]), 129600)
```

Reply:
done